src/fsrs/convert.ts

```diff
@@ -47,8 +47,12 @@ export class TypeConvert {
     throw new Error(`Invalid state:[${value}]`)
   }
   static time(value: unknown): Date {
-    if (typeof value === 'object' && value instanceof Date) {
-      return value
+    if (
+      typeof value === 'object' &&
+      value !== null &&
+      !Number.isNaN(Date.parse(value as unknown as string))
```

Comment:
There’s a possible parsing mismatch case:

```javascript
> const test=Object.assign(1755047663017,{toString:()=>'mock'})
undefined
> typeof test
'object'
> new Date(test)
2025-08-13T01:14:23.017Z
> Date.parse(test)
NaN
```

Here, new Date(test) works because the object’s valueOf() returns a number (timestamp), but Date.parse(test) fails since it uses the toString() result, which is not a valid date string.

Reply:
```javascript
> const test2=Object.assign(1755047663017,{})
undefined
> typeof test2
'object'
> Date.parse(test2)
NaN
> new Date(test2)
2025-08-13T01:14:23.017Z
```

Reply:
@ishiko732  All of these currently throw an error, I believe. This PR will allow the third one to pass, while still throwing the first 2 tests.

Do you reckon we should also allow the first two to pass as well? Something like:

```js
const date = new Date(value);

if (
  typeof value === 'object' &&
  value !== null &&
  (
    !Number.isNaN(Date.parse(value)) ||
    !Number.isNaN(date.getTime())
  )
) {
  return date;
}
```

Reply:
> Do you reckon we should also allow the first two to pass as well?

I think the first two tests should also pass, since they can validly produce a Date instance.

Reply:
> Something like

You could simplify that check to:
```js
!Number.isNaN(Date.parse(value) || +date)
```
This way you only call Number.isNaN once, while still covering both the string-parsable case and the numeric timestamp case.

Reply:
I‘m not sure how big of a deal that is, you can already get invalid dates from this converter by passing to it something like `new Date("invalid-date")`, although after this PR that will now fail.